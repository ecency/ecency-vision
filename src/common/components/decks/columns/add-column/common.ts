export interface SettingsProps {
  deckKey: number;
}

export interface UsernameDataItem {
  name: string;
  description?: string;
  tag?: string;
}
