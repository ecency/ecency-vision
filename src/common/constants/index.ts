export * from "./faq-keys-general";
