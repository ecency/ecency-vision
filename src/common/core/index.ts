export * from "./react-query";
export * from "./caches";
