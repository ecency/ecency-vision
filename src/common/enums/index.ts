export * from "./notify-types";
export * from "./error-types";
