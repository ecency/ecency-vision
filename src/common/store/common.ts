export enum CommonActionTypes {
  LOCATION_CHANGE = '@@router/LOCATION_CHANGE',
}

export interface LocationChangeAction {
  type: CommonActionTypes.LOCATION_CHANGE;
  payload: {
    location: {
      pathname: string;
    };
    action?: 'PUSH' | 'POP';
  };
}

export const locationChangeAct = (pathname: string): LocationChangeAction => {
  return {
    type: CommonActionTypes.LOCATION_CHANGE,
    payload: {
      location: {
        pathname,
      },
    },
  };
};
