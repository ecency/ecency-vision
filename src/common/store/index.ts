import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";
import { createBrowserHistory, createMemoryHistory, History } from "history";

import isEqual from "react-fast-compare";

import global from "./global";
import dynamicProps from "./dynamic-props";
import trendingTags from "./trending-tags";
import entries from "./entries";
import accounts from "./accounts";
import communities from "./communities";
import transactions from "./transactions";
import users from "./users";
import activeUser from "./active-user";
import reblogs from "./reblogs";
import discussion from "./discussion";
import ui from "./ui";
import subscriptions from "./subscriptions";
import notifications from "./notifications";
import points from "./points";
import signingKey from "./signing-key";
import entryPinTracker from "./entry-pin-tracker";
import persistentPageScroll from "./persistent-page-scroll";
import deck from "./deck";

import filterTagExtract from "../helper/filter-tag-extract";

let reducers = {
  global,
  dynamicProps,
  trendingTags,
  entries,
  accounts,
  communities,
  transactions,
  users,
  activeUser,
  reblogs,
  discussion,
  ui,
  subscriptions,
  notifications,
  points,
  signingKey,
  entryPinTracker,
  deck,
  persistentPageScroll
};

export let history: History | undefined;

// create browser history on client side
if (typeof window !== "undefined") {
  if (window.location.href.startsWith("file://")) {
    // electron
    history = createMemoryHistory();
  } else {
    // web
    history = createBrowserHistory();
  }

  // We need a customised history object since history pushes new state for same path.
  // See: https://github.com/ReactTraining/history/issues/470
  // We don't want LOCATION_CHANGE triggered for same path because of structure of out "entries" reducer.

  // get ref of current push function
  const _push = history.push;

  let prevPath: string = history.location.pathname;
  // update previous path once history change
  history.listen((location) => {
    prevPath = location.pathname;
  });

  // create a new push function that compares new path and previous path.
  history.push = (pathname: History.Path, state: History.LocationState = {}) => {
    // compare filter & tag resolution of paths
    // this control required because "/" == "/hot" and "/@username" == "/@username/posts"
    const ftCur = filterTagExtract(pathname);
    const ftPrev = filterTagExtract(prevPath);

    if (ftCur && ftPrev) {
      if (isEqual(ftCur, ftPrev)) {
        return;
      }
    }

    // simple path compare
    if (pathname === prevPath) {
      return;
    }
    _push(pathname.includes("//") ? "/" : pathname, state);
  };

  // scroll to top on every push action
  history.listen((location, action) => {
    if (action === "PUSH") {
      // Don't scroll to top with anchor links
      if (history!.location.hash !== "") {
        return;
      }

      setTimeout(() => {
        window.scrollTo({
          top: 0
        });
      }, 100);
    }
  });

  // @ts-ignore
  reducers = { router: connectRouter(history), ...reducers };
}

const rootReducer = combineReducers(reducers);

export default rootReducer;

export type AppState = ReturnType<typeof rootReducer>;
