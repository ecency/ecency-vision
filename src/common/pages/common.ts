import { AnyAction, bindActionCreators, Dispatch } from "redux";
import { AppState } from "../store";

import { History, Location } from "history";
import { Global } from "../store/global/types";
import { User } from "../store/users/types";
import { Account, Accounts } from "../store/accounts/types";
import { Community, Communities } from "../store/communities/types";
import { TrendingTags } from "../store/trending-tags/types";
import { ActiveUser } from "../store/active-user/types";
import { ToggleType, UI } from "../store/ui/types";
import { NotificationFilter, Notifications } from "../store/notifications/types";
import { Subscription } from "../store/subscriptions/types";
import { DynamicProps } from "../store/dynamic-props/types";
import { Entries, Entry } from "../store/entries/types";
import { Reblogs } from "../store/reblogs/types";
import { Discussion as DiscussionType, SortOrder } from "../store/discussion/types";
import { Transactions, OperationGroup } from "../store/transactions/types";
import { Points } from "../store/points/types";
import { EntryPinTracker } from "../store/entry-pin-tracker/types";

import {
  toggleTheme,
  hideIntro,
  toggleListStyle,
  dismissNewVersion,
  muteNotifications,
  unMuteNotifications,
  setCurrency,
  setLang,
  setNsfw,
  setLastIndexPath,
  setShowSelfVote,
  setShowRewardSplit,
  setLowRewardThreshold
} from "../store/global";
import { fetchTrendingTags } from "../store/trending-tags";
import { updateSubscriptions } from "../store/subscriptions";
import { fetchEntries, addEntry, updateEntry, invalidateEntries } from "../store/entries";
import {
  fetchDiscussion,
  sortDiscussion,
  resetDiscussion,
  updateReply,
  addReply,
  deleteReply
} from "../store/discussion";
import { addAccount } from "../store/accounts";
import { addCommunity } from "../store/communities";
import { fetchTransactions, resetTransactions } from "../store/transactions";
import { addUser, deleteUser } from "../store/users";
import { setActiveUser, updateActiveUser } from "../store/active-user";
import { toggleUIProp } from "../store/ui";
import { fetchReblogs, addReblog, deleteReblog } from "../store/reblogs";
import {
  fetchNotifications,
  fetchUnreadNotificationCount,
  setNotificationsFilter,
  markNotifications,
  updateNotificationsSettings,
  fetchNotificationsSettings,
  setNotificationsSettingsItem
} from "../store/notifications";
import { fetchPoints, resetPoints } from "../store/points";
import { setSigningKey } from "../store/signing-key";
import { trackEntryPin, setEntryPin } from "../store/entry-pin-tracker";
import {
  createDeck,
  deleteDeck,
  fetchDeckData,
  loadDeckFromStorage,
  reorderDecks,
  setDeckDataFilters,
  setDeckUpdateInterval
} from "../store/deck";
import { savePageScroll } from "../store/persistent-page-scroll";

export interface PageProps {
  history: History;
  location: Location;

  global: Global;
  toggleTheme: () => void;
  hideIntro: () => void;
  toggleListStyle: (view: string | null) => void;
  dismissNewVersion: () => void;
  muteNotifications: () => void;
  unMuteNotifications: () => void;
  setCurrency: (currency: string, rate: number, symbol: string) => void;
  setLang: (lang: string) => void;
  setNsfw: (value: boolean) => void;
  setLastIndexPath: (path: string | null) => void;
  setShowSelfVote: (value: boolean) => void;
  setShowRewardSplit: (value: boolean) => void;
  setLowRewardThreshold: (value: number) => void;

  dynamicProps: DynamicProps;

  trendingTags: TrendingTags;
  fetchTrendingTags: () => void;

  subscriptions: Subscription[];
  updateSubscriptions: (list: Subscription[]) => void;

  entries: Entries;
  fetchEntries: (what: string, tag: string, more: boolean) => void;
  addEntry: (entry: Entry) => void;
  updateEntry: (entry: Entry) => void;
  invalidateEntries: (groupKey: string) => void;

  discussion: DiscussionType;
  fetchDiscussion: (parent_author: string, parent_permlink: string) => void;
  sortDiscussion: (order: SortOrder) => void;
  resetDiscussion: () => void;
  updateReply: (reply: Entry) => void;
  addReply: (reply: Entry) => void;
  deleteReply: (reply: Entry) => void;

  accounts: Accounts;
  addAccount: (data: Account) => void;

  communities: Communities;
  addCommunity: (data: Community) => void;

  transactions: Transactions;
  fetchTransactions: (
    username: string,
    group?: OperationGroup | "",
    start?: number,
    limit?: number
  ) => void;
  resetTransactions: () => void;

  users: User[];
  addUser: (user: User) => void;
  deleteUser: (username: string) => void;

  activeUser: ActiveUser | null;
  setActiveUser: (username: string | null) => void;
  updateActiveUser: (data?: Account) => void;

  ui: UI;
  toggleUIProp: (what: ToggleType) => void;

  reblogs: Reblogs;
  fetchReblogs: () => void;
  addReblog: (author: string, permlink: string) => void;
  deleteReblog: (author: string, permlink: string) => void;

  notifications: Notifications;
  fetchNotifications: (since: string | null) => void;
  fetchUnreadNotificationCount: () => void;
  setNotificationsFilter: (filter: NotificationFilter | null) => void;
  markNotifications: (id: string | null) => void;
  updateNotificationsSettings: typeof updateNotificationsSettings;
  fetchNotificationsSettings: typeof fetchNotificationsSettings;
  setNotificationsSettingsItem: typeof setNotificationsSettingsItem;

  points: Points;
  fetchPoints: (username: string, filter?: number) => void;
  resetPoints: () => void;

  signingKey: string;
  setSigningKey: (key: string) => void;

  deck: any;
  persistentPageScroll: any;
  entryPinTracker: EntryPinTracker;
  trackEntryPin: (entry: Entry) => void;
  setEntryPin: (entry: Entry, pin: boolean) => void;
  createDeck: typeof createDeck;
  fetchDeckData: typeof fetchDeckData;
  loadDeckFromStorage: typeof loadDeckFromStorage;
  deleteDeck: typeof deleteDeck;
  reorderDecks: typeof reorderDecks;
  savePageScroll: typeof savePageScroll;
  hiveSignerApp: string;
}

export const pageMapStateToProps = (state: AppState) => ({
  ...state
});

export const pageMapDispatchToProps = (dispatch: Dispatch<AnyAction>) =>
  bindActionCreators(
    {
      toggleTheme,
      hideIntro,
      toggleListStyle,
      muteNotifications,
      unMuteNotifications,
      setCurrency,
      setLang,
      setNsfw,
      setLastIndexPath,
      setShowSelfVote,
      setShowRewardSplit,
      setLowRewardThreshold,
      dismissNewVersion,
      fetchTrendingTags,
      updateSubscriptions,
      fetchEntries,
      addEntry,
      updateEntry,
      invalidateEntries,
      fetchDiscussion,
      sortDiscussion,
      resetDiscussion,
      updateReply,
      addReply,
      deleteReply,
      addAccount,
      addCommunity,
      fetchTransactions,
      resetTransactions,
      addUser,
      deleteUser,
      setActiveUser,
      updateActiveUser,
      toggleUIProp,
      addReblog,
      deleteReblog,
      fetchReblogs,
      fetchNotifications,
      fetchUnreadNotificationCount,
      setNotificationsFilter,
      markNotifications,
      fetchPoints,
      resetPoints,
      setSigningKey,
      trackEntryPin,
      setEntryPin,
      createDeck,
      fetchDeckData,
      loadDeckFromStorage,
      deleteDeck,
      reorderDecks,
      savePageScroll,
      setDeckUpdateInterval,
      setDeckDataFilters,
      updateNotificationsSettings,
      fetchNotificationsSettings,
      setNotificationsSettingsItem
    },
    dispatch
  );
