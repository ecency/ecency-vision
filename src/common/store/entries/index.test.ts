import reducer, {
  initialState,
  fetchAct,
  fetchErrorAct,
  fetchedAct,
  invalidateAct,
  updateAct
} from './index';

import { locationChangeAct } from '../common';

let state = initialState;

it('1- default state', () => {
  expect(state).toMatchSnapshot();
});

it('2- location change. should create a group with default filter.', () => {
  state = reducer(state, locationChangeAct('/'));
  expect(state).toMatchSnapshot();
});

it('3- location change. should create a group with the filter and tag given.', () => {
  state = reducer(state, locationChangeAct('/trending/bitcoin'));
  expect(state).toMatchSnapshot();
});

it('4- location change with invalid filter. state should state same.', () => {
  state = reducer(state, locationChangeAct('/trendinx/bitcoin'));
  expect(state).toMatchSnapshot();
});

it('5- start fetching. should toggle "loading"', () => {
  const act = fetchAct('hot');
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('6- fetch error. should toggle "loading" and add error object to the group.', () => {
  const act = fetchErrorAct('hot', 'fetch error');
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('7- start fetching.', () => {
  const act = fetchAct('trending-bitcoin');
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('8- fetched.', () => {
  const entries = [
    {
      author: 'lorem',
      permlink: 'lorem-ipsum-dolor-sit-amet',
      payout: 1
    },
    {
      author: 'consectetur',
      permlink: 'consectetur-adipiscing elit-in-vel-enim-libero',
      payout: 1
    },
    {
      author: 'nam',
      permlink: 'tempor-nibh-eget-enim-vulputate',
      payout: 1
    },
    {
      author: 'vivamus',
      permlink: 'vivamus-ornare-at-tortor-eu-vestibulum',
      payout: 1
    }
  ];

  // @ts-ignore
  const act = fetchedAct('trending-bitcoin', entries, true);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('9- start fetching again (simulates user scrolling down)', () => {
  const act = fetchAct('trending-bitcoin');
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('10- fetched. should ignore posts already in store.', () => {
  const entries = [
    {
      author: 'vivamus',
      permlink: 'vivamus-ornare-at-tortor-eu-vestibulum'
    },
    {
      author: 'donec',
      permlink: 'donec-bibendum-tortor-ultricies',
      payout: 1
    }
  ];

  // @ts-ignore
  const act = fetchedAct('trending-bitcoin', entries, false);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('11- location change for an extra group', () => {
  state = reducer(state, locationChangeAct('/created/bitcoin'));
  expect(state).toMatchSnapshot();
});

it('12- fetched another group.', () => {
  const entries = [
    {
      author: 'nam',
      permlink: 'tempor-nibh-eget-enim-vulputate',
      payout: 1
    }
  ];

  // @ts-ignore
  const act = fetchedAct('created-bitcoin', entries, false);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('13- update a particular entry.', () => {
  const entry = {
    author: 'nam',
    permlink: 'tempor-nibh-eget-enim-vulputate',
    payout: 1.2
  };

  // @ts-ignore
  const act = updateAct(entry);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('14- invalidate. should reset the group.', () => {
  const act = invalidateAct('trending-bitcoin');
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
