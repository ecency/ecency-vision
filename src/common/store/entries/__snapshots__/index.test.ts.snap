// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
}
`;

exports[`2- location change. should create a group with default filter. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`3- location change. should create a group with the filter and tag given. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`4- location change with invalid filter. state should state same. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`5- start fetching. should toggle "loading" 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`6- fetch error. should toggle "loading" and add error object to the group. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`7- start fetching. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
}
`;

exports[`8- fetched. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": false,
    "sid": true,
  },
}
`;

exports[`9- start fetching again (simulates user scrolling down) 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": true,
    "sid": true,
  },
}
`;

exports[`10- fetched. should ignore posts already in store. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
      Object {
        "author": "donec",
        "payout": 1,
        "permlink": "donec-bibendum-tortor-ultricies",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": false,
    "sid": false,
  },
}
`;

exports[`11- location change for an extra group 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "created-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": true,
    "sid": "",
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
      Object {
        "author": "donec",
        "payout": 1,
        "permlink": "donec-bibendum-tortor-ultricies",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": false,
    "sid": false,
  },
}
`;

exports[`12- fetched another group. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "created-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": false,
    "sid": false,
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
      Object {
        "author": "donec",
        "payout": 1,
        "permlink": "donec-bibendum-tortor-ultricies",
      },
    ],
    "error": null,
    "hasMore": undefined,
    "loading": false,
    "sid": false,
  },
}
`;

exports[`13- update a particular entry. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "created-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "nam",
        "payout": 1.2,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
    ],
    "error": null,
    "loading": false,
    "sid": false,
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "lorem",
        "payout": 1,
        "permlink": "lorem-ipsum-dolor-sit-amet",
      },
      Object {
        "author": "consectetur",
        "payout": 1,
        "permlink": "consectetur-adipiscing elit-in-vel-enim-libero",
      },
      Object {
        "author": "nam",
        "payout": 1.2,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
      Object {
        "author": "vivamus",
        "payout": 1,
        "permlink": "vivamus-ornare-at-tortor-eu-vestibulum",
      },
      Object {
        "author": "donec",
        "payout": 1,
        "permlink": "donec-bibendum-tortor-ultricies",
      },
    ],
    "error": null,
    "loading": false,
    "sid": false,
  },
}
`;

exports[`14- invalidate. should reset the group. 1`] = `
Object {
  "__manual__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "__promoted__": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "created-bitcoin": Object {
    "entries": Array [
      Object {
        "author": "nam",
        "payout": 1.2,
        "permlink": "tempor-nibh-eget-enim-vulputate",
      },
    ],
    "error": null,
    "loading": false,
    "sid": false,
  },
  "hot": Object {
    "entries": Array [],
    "error": "fetch error",
    "hasMore": false,
    "loading": false,
    "sid": "",
  },
  "trending-bitcoin": Object {
    "entries": Array [],
    "error": null,
    "hasMore": false,
    "loading": false,
    "sid": false,
  },
}
`;
